src/code/z_camera.c

```diff
@@ -543,323 +868,7415 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE79C.s")
+    pitchNear = CAM_DEG_TO_BINANG(RADF_TO_DEGF(func_80086B30(floorYDiffNear, nearDist)));
+    pitchFar = CAM_DEG_TO_BINANG(RADF_TO_DEGF(func_80086B30(floorYDiffFar, farDist)));
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE930.s")
+    return pitchNear + pitchFar;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CEAD8.s")
+f32 func_800CCCEC(Camera* camera, s16 reset) {
+    static s32 D_801B9E5C = 0;
+    static f32 D_801B9E60 = 0.0f;
+    Vec3f offsetForwards;
+    Vec3f angledOffsetForwards;
+    PosRot focalActorPosRot;
+    CameraCollision camCollision;
+    f32 forwardsUnitZ;
+    f32 focalActorHeight;
+    f32 forwardsUnitX;
+    f32 distResult;
+    s16 yawForwardsOffset;
+    f32 distResultAdj;
+
+    focalActorHeight = Camera_GetFocalActorHeight(camera);
+
+    if (reset) {
+        D_801B9E5C = 0;
+        D_801B9E60 = 0.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CEC38.s")
+    focalActorPosRot = camera->focalActorPosRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CED90.s")
+    forwardsUnitX = Math_SinS(focalActorPosRot.rot.y);
+    forwardsUnitZ = Math_CosS(focalActorPosRot.rot.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Noop.s")
+    offsetForwards.x = focalActorPosRot.pos.x + (30.0f * forwardsUnitX);
+    offsetForwards.y = focalActorPosRot.pos.y + focalActorHeight;
+    offsetForwards.z = focalActorPosRot.pos.z + (30.0f * forwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal1.s")
+    camCollision.pos.x = focalActorPosRot.pos.x + (12.0f * forwardsUnitX);
+    camCollision.pos.y = offsetForwards.y;
+    camCollision.pos.z = focalActorPosRot.pos.z + (12.0f * forwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal2.s")
+    if ((camera->play->state.frames % 2) != 0) {
+        // Turn and project 65.92 degrees left
+        yawForwardsOffset = focalActorPosRot.rot.y + DEG_TO_BINANG(65.92f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal3.s")
+        angledOffsetForwards.x = (Math_SinS(yawForwardsOffset) * 50.0f) + offsetForwards.x;
+        angledOffsetForwards.y = offsetForwards.y;
+        angledOffsetForwards.z = (Math_CosS(yawForwardsOffset) * 50.0f) + offsetForwards.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal4.s")
+        if ((Camera_BgCheckInfo(camera, &angledOffsetForwards, &camCollision) != 0) &&
+            Camera_CheckOOB(camera, &offsetForwards, &angledOffsetForwards)) {
+            // Always going to result in 28.0f?
+            distResult = OLib_Vec3fDistXZ(&offsetForwards, &camCollision.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal0.s")
+            // (-D_801B9E60 < (50.0f - distResult))
+            if ((D_801B9E5C != 2) || !(-D_801B9E60 >= (50.0f - distResult))) {
```

Comment:
```suggestion
            if (!((D_801B9E5C == 2) && (-D_801B9E60 >= (50.0f - distResult)))) {
```
Up to you, but this seems cleaner to me.